### PR TITLE
Validators for IAM

### PR DESCRIPTION
**Pull request by dsoffiantini**
## Description

Adds a set of griffon validators for In App Messaging events and configuration

## Motivation and Context

With the launch of IAM, we want to provide users feedback on any errors they may encounter with their app.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: plugins/aep-iam-configuration/index.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  return configEvents.some(hasAllConfig)
+    ? {
+        message: 'IAM has been configured correctly',
+        errors: [],
+        status: 'valid'
+      }
+    : {
+        message:
+          'Did not detect the required configuration values. Please ensure the following values have been configured in launch: experienceCloud.org, edge.configId, and messaging.eventDataset.',
+        errors: configEvents.map((event) => event.uuid),
```
Comment (dgrossen) [inline plugins/aep-iam-configuration/index.ts:48]: `errors` is only supported for backwards compatibility and should be replaced with `events`

File: plugins/aep-iam-configuration/index.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        status: 'invalid'
```
Comment (dgrossen) [inline plugins/aep-iam-configuration/index.ts:49]: I don't think this is a valid enumerated value for `status`

File: plugins/aep-iam-configuration/index.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      };
```
Comment (dgrossen) [inline plugins/aep-iam-configuration/index.ts:50]: `result` enumerated value is required for validators

https://github.com/adobe/griffon-validation-plugins/wiki/Writing-a-Validation-Function#returning-validation-results

**Review by dgrossen — CHANGES_REQUESTED** [review]
Looks like in this PR and in other validators we aren't returning the expected results object. These validators will fail when run in the validation manager (bootes)

https://github.com/adobe/griffon-validation-plugins/wiki/Writing-a-Validation-Function#returning-validation-results

**Comment by dgrossen** [comment]
The following validators are returning an incompatible results object:

https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/aep-messaging-config/index.ts
https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/aep-messaging-installed/index.ts
https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/aep-messaging-push-on-platform/index.ts
https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/aep-messaging-pushid/index.ts
https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/aep-optimize-peer-dependencies/index.ts
https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/aep-optimize-prerequisites/index.ts
https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/edge-config/index.ts
https://github.com/adobe/griffon-validation-plugins/blob/master/plugins/edge-installed/index.ts

**Comment by amadt** [comment]
@dgrossen: Sounds like we should flesh out the `ValidationPluginResult` type?

**Comment by dgrossen** [comment]
@amadt I believe we already did back in September. The manager only accepts `loading`, `invalid auth`, and `prerequisite missing`. `invalid` and `valid` are superfluous to `not matched` and `matched`

https://git.corp.adobe.com/dms-mobile/bootes/blob/master/packages/validation-plugin-manager/src/sagas/validate.events.saga.js#L46

**Comment by dsoffiantini** [comment]
@dgrossen Updated out typescript definition to match what the platform expects. I fixed up the plugins minus the optimize ones, which are fixed on my other pr :)

File: plugins/aep-iam-request/index.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  return errors.length
+    ? {
+        result: 'invalid',
+        message:
+          'There are IAM request scopes that did not have a corresponding response',
+        errors
+      }
+    : {
+        result: 'valid',
```
Comment (dgrossen) [inline plugins/aep-iam-request/index.ts:59]: invalid result enum

`matched` | `not matched` | `unknown`

File: plugins/aep-iam-response-events/index.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        } ${
+          !!invalidContent.length &&
+          `There are responses missing IAM rules in the payload.`
+        }`,
+        result: 'invalid',
+        errors
+      }
+    : {
+        message: 'Valid! All IAM Response Events are correct.',
+        result: 'valid'
```
Comment (dgrossen) [inline plugins/aep-iam-response-events/index.ts:104]: invalid result enum

**Review by dgrossen — CHANGES_REQUESTED** [review]
a few invalid result enums

File: plugins/aep-iam-response-events/index.ts (hunk 5 above)
Comment (dsoffiantini) [reply plugins/aep-iam-response-events/index.ts]: @dgrossen Think I got them all :)

**Review by dgrossen — APPROVED** [review]

**Pull request merged**